### PR TITLE
fix: Updating dependencies and fixes

### LINKED ISSUE #119: Dated library
Kia ora team, we are hoping to switch over from Sentry to Raygun, but are bumping into the fact that this Node library seems to be very dated and not really up to speed with modern best practices. The guide for using this library that the Sentry app links to has been written in 2013, and the code examples show it.

Are there any plans to modernise this library in the near future?

It will be very hard for us to rewrite our current Sentry error handling boilerplate code to migrate this to the Raygun library, which still uses callbacks, and it seems we will lose a lot of vital functionality and ease of implementation in the process.

We really would like to migrate, but this seems to be a show stopper 😢

### PR DESCRIPTION
**Pull request by miquelbeltran**
This PR upgrades dependencies in the root project

- Used `node-version-check` to review and perform upgrades.
- Recreated the `package-lock.json` to fix dependency issues.
- fix mix up between `express` `Request` type and Raygun's `RequestParams` type.
- import `deepEqual` from the `assert` package.
- fixes in the `tap` test command
- replaces the removed `assert` calls from the `tap` project with the `tap` equivalent.

TODO in a different PR: 

- Update dependencies for the example projects.
- Replace `tslint` with `eslint`.
- Setup `eslint` in GitHub Actions and fix all static analysis warnings.

**After fixing dependencies and tests:**

```bash
$ npm ci

npm WARN deprecated tslint-react@5.0.0: tslint-react is deprecated along with TSLint
npm WARN deprecated tslint@6.1.3: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.

> raygun@0.13.2 prepare
> tsc


added 526 packages, and audited 527 packages in 3s

83 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

```bash
$ npm test


                       
  🌈 TEST COMPLETE 🌈  
                       

Asserts:  217 pass  0 fail  217 of 217 complete
Suites:     7 pass  0 fail      7 of 7 complete

# No coverage generated
# { total: 217, pass: 217 }
# time=13397.089ms
```

### Affected issues

- Closes #119

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 yarn.lock
 
 build/
+.tap/
```
Comment (miquelbeltran) [inline .gitignore:6]: temporal test result files, ignored so they don't get commited

File: package.json [hunk 2]
```diff
@@ -34,32 +34,32 @@
     "tslint": "tslint --fix --project .",
     "prettier": "prettier --write lib/*.ts test/*.js",
     "prepare": "tsc",
-    "test": "tap --no-ts --node-arg=-r --node-arg=ts-node/register --no-coverage test/*_test.js"
```
Comment (miquelbeltran) [inline package.json:37]: `--no-ts` is no longer a valid param of `tap`. `--no-coverage` was replaced by `--disable-coverage`

File: package.json [hunk 3]
```diff
@@ -34,32 +34,32 @@
...
+    "http-terminator": "^3.2.0",
+    "jshint": "^2.13.6",
+    "nock": "~13",
+    "prettier": "^3.2.5",
+    "semver": "^7.6.0",
+    "tap": "^18.7.2",
+    "ts-node": "^10.9.2",
     "tslint": "^6.1.1",
     "tslint-config-prettier": "^1.18.0",
-    "tslint-react": "^4.2.0",
```
Comment (miquelbeltran) [inline package.json:53]: Note that `tslint` is deprecated and should be replaced with `eslint`, I plan to do that in the future.

File: test/raygun_batch_lifecycle_test.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ test("batch transport doesn't keep processes alive", async (t) => {
           return reject(error);
         }
 
-        t.assert(process.exitCode === 0);
```
Comment (miquelbeltran) [inline test/raygun_batch_lifecycle_test.js:15]: `assert` is no longer part of `tap`, replace by methods like `ok()` or `equal()`

File: test/raygun.messageBuilder_test.js [hunk 5]
```diff
@@ -404,7 +405,7 @@ test("custom tags", function (t) {
     builder.setTags(["a", "bb", "c"]);
     var message = builder.build();
 
-    tt.deepEqual(message.details.tags, ["a", "bb", "c"]);
```
Comment (miquelbeltran) [inline test/raygun.messageBuilder_test.js:407]: `deepEqual()` is not part of `tap`, replaced by the `deepEqual` from the `assert` module.

File: lib/raygun.ts [hunk 6]
```diff
@@ -133,7 +133,7 @@ class Raygun {
     return this;
   }
 
-  user(req?: Request): RawUserData | null {
+  user(req?: RequestParams): RawUserData | null {
```
Comment (miquelbeltran) [inline lib/raygun.ts:136]: The original code mixed the `Request` type from Express with the internal `RequestParams` type, after upgrading dependencies the "compiler" didn't like that

**Review by TheRealAgentK — APPROVED** [review]
LGTM, left most of your comments unresolved (apart from one that is now #145) so that others can get information on changes.

File: package-lock.json
[the remark this answers is not in the document]
Comment (miquelbeltran) [reply package-lock.json:1]: Yes, I even had to delete and recreate it to resolve some issues.

**Comment by TheRealAgentK** [comment]
@miquelbeltran Probably a good idea to create separate issues for

> TODO in a different PR:
> 
> Update dependencies for the example projects.
> Replace tslint with eslint.
> Setup eslint in GitHub Actions and fix all static analysis warnings.

(second one I've done already when I went through this PR's changes)

**Pull request merged**